src/routes/(app)/+page.server.ts

```diff
@@ -1,28 +1,23 @@
 import type { RequestEvent } from '@sveltejs/kit';
 import type { PageServerLoad } from './$types';
+import type { stringsSchema } from '$lib/locale.store';
+import type { z } from 'zod';
 
 import { getLocale } from '$lib/_utilities/i18n.utilities';
 
-// Supported locales
-import ca_en from '$locales/ca/en/index.json';
-import ca_fr from '$locales/ca/fr/index.json';
-import us_en from '$locales/us/en/index.json';
-import us_es from '$locales/us/es/index.json';
-
-const locales = {
-  ca_en,
-  ca_fr,
-  us_en,
-  us_es,
-};
-
 export const load: PageServerLoad = async (event: RequestEvent) => {
   const userLocale = event.request.headers.get('accept-language') || 'en-US';
+  const locale = getLocale(userLocale, '/');
+  const [country, lang] = locale.split('/');
+
+  let localeContent: { default: z.infer<typeof stringsSchema> };
 
-  // TODO: Reworking typing here to index the locales object
-  const locale = getLocale(userLocale, '_') as 'ca_en' | 'ca_fr' | 'us_en' | 'us_es';
+  try {
+    localeContent = await import(`../../locales/${country}/${lang}/index.json`);
+  } catch (err) {
+    console.error(`User locale content for ${userLocale} was not found.`);
+    localeContent = await import(`../../locales/us/en/index.json`);
+  }
```

Comment:
Finally figured out how to do async imports properly \o/

Reply:
Nice! I think this change also means that we are no longer hard coding locales, instead letting the naming of the files define the locales available - right?

Reply:
That is correct.